core/src/main/java/org/apache/iceberg/puffin/PuffinFormat.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.puffin;
+
+import io.airlift.compress.Compressor;
+import io.airlift.compress.zstd.ZstdCompressor;
+import io.airlift.compress.zstd.ZstdDecompressor;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableTable;
+import org.apache.iceberg.relocated.com.google.common.collect.Table;
+import org.apache.iceberg.util.ByteBuffers;
+
+final class PuffinFormat {
+  private PuffinFormat() {
+  }
+
+  enum Flag {
+    FOOTER_PAYLOAD_COMPRESSED(0, 0),
+    /**/;
+
+    private static final Table<Integer, Integer, Flag> BY_BYTE_AND_BIT = Stream.of(values())
```

Comment:
Rather than adding a new Guava class, I think you could just use a `Map<Pair<Integer, Integer>, Flag>`. The lookup is hidden behind a static method anyway, so there isn't much of a cost to using `Pair`.

Reply:
awesome idea!